build/protobuf.mk

```diff
@@ -39,8 +39,8 @@ GOGOPROTO_PATH  := $(GOGOPROTO_ROOT):$(GOGOPROTO_ROOT)/protobuf
 
 COREOS_PATH := $(GITHUB_ROOT)/coreos
 
-GO_PROTOS  := $(sort $(shell find $(REPO_ROOT) -not -path '*/.*' -name *.proto -type f))
-GO_SOURCES  := $(GO_PROTOS:%.proto=%.pb.go)
+GO_PROTOS := $(addprefix $(REPO_ROOT)/, $(sort $(shell cd $(REPO_ROOT) && git ls-files --exclude-standard --cached --others -- '*.proto')))
```

Comment:
Nice, TIL.
On Dec 14, 2015 18:41, "Ben Darnell" <notifications@github.com> wrote:

> In build/protobuf.mk
> <https://github.com/cockroachdb/cockroach/pull/3436#discussion_r47577134>:
>
> > @@ -39,8 +39,8 @@ GOGOPROTO_PATH  := $(GOGOPROTO_ROOT):$(GOGOPROTO_ROOT)/protobuf
> >
> >  COREOS_PATH := $(GITHUB_ROOT)/coreos
> >
> > -GO_PROTOS  := $(sort $(shell find $(REPO_ROOT) -not -path '*/.*' -name *.proto -type f))
> > -GO_SOURCES  := $(GO_PROTOS:%.proto=%.pb.go)
> > +GO_PROTOS := $(addprefix $(REPO_ROOT)/, $(sort $(shell cd $(REPO_ROOT) && git ls-files --exclude-standard --cached --others -- '*.proto')))
>
> Minor tip: you can pass git -C $(REPO_ROOT) instead of running cd in a
> subshell.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/3436/files#r47577134>.
>